### ISSUE TITLE
[EPIC][Bonus] Print functionality

### ISSUE THREAD

**Issue by alxzak** [kirbydesign/designsystem#1544]
The old netbank have a print and export functionality.
We have to have the same funtionality supported in the new netbank. 

**Describe the solution you'd like**
Kirby team can help develop the print style sheet and buttons, but the backend must generate content. 

Please contact Christian Lüdemann and/or Mikael Kristoffersen for more information.

### Kick Off:

- [ ] Ensure the enhancement is `UX refined` and aligned with UX 
_The component and/or enhancement should be published and available in the [Kirby Styleguide on Zeplin](https://zpl.io/258pXGj)_
- [ ] Ensure the enhancement has been `Tech refined` with @kirbydesign/kirby-guild and this issue is updated with a clear implementation description  
_This issue should be in the [Ready to do](https://github.com/kirbydesign/designsystem/projects/1#column-4590936) column of the [Kirby kan-ban board](https://github.com/kirbydesign/designsystem/projects/1) before starting implementation)_
- [ ] Assign yourself to this issue and move it to the [In progress](https://github.com/kirbydesign/designsystem/projects/1#column-4590937) column of the [Kirby kan-ban board](https://github.com/kirbydesign/designsystem/projects/1)

### Code:

- [ ] Create Feature Branch from [master branch](https://github.com/kirbydesign/designsystem/tree/master)
- [ ] Create a draft implementation and push to Github
- [ ] Ask a member of @kirbydesign/kirby-guild for a WIP review by creating a draft Pull Request
- [ ] Implement unit tests
- [ ] Update Cookbook Examples and Showcase, i.e. see [Radio](https://cookbook.kirby.design/home/showcase/radio)  
_Also remember to add any relevant new API documentation_

### Review:

- UX review:
  - [ ] Ensure implementation is correct in relation to the UX design and the [Kirby Styleguide on Zeplin](https://zpl.io/258pXGj)
  - [ ] With UX agree on the version of the implementation
- Code review:
  - [ ] Open a pull request (or mark the existing draft PR as `Ready for review`) and ask @kirbydesign/kirby-guild for a review  
_Remember to add `closes #issueno` to the description of the PR._
  - [ ] Once approved, merge feature branch/PR to master
- [ ] Ask a member of @kirbydesign/kirby-guild to add a link to component showcase from Kirby Component Status and update the version number

:tada: Celebrate

**Comment by alxzak** [comment]
Team 2, 3 and 5 are stakeholders. 
@alxzak invites to a kickoff, where the different needs are discussed and the solution takes form.

**Comment by jkaltoft** [comment]
## Meeting minutes (partly), June 8 2021

### Print

* Stakeholders prefer a solution where they are able to print using <kbd>CTRL</kbd>/<kbd>⌘</kbd> + <kbd>P</kbd> but also to print more than what is visible on screen (related to *virtual scroll*)
* Stakeholders will provide specific examples of pages/views that must be "print friendly"
    - *Posteringsliste*
    - Focus on design/layout later
* Kirby will look into raising the base level for "print friendliness" in general to the best possible and meaningful extent
* *Virtual scroll* is expected to introduce some challenges. Kirby will look into possible solutions.

### PDF

* Can the print stylesheet(s) be used directly for PDF generation?

### Export to CSV/Excel

* NOOP()

**Comment by alxzak** [comment]
![Slice_Konto_Postering.png](https://images.zenhubusercontent.com/602ac85e6b87087753686a4f/eb908910-8d4f-4b36-8018-22437440b3f9)![Slice_Eksporter_Kontobevægelser.png](https://images.zenhubusercontent.com/602ac85e6b87087753686a4f/86bdccb3-2266-4f49-9a7e-9aa105480b84)

**Comment by alxzak** [comment]
Tilde Vinge Hem  [3:56 PM]
Hej Alex. Her lige lidt info fra team 5 ang Print/Eksport. Vi vil gerne bede jer om at tage udgangspunkt i posteringslisten og posteringsdetaljesiden samt det flow vi har skitseret. nedenfor finder du billeder af det hele. Ang look and feel kan I række ud til Markus vores UX'er og ang teknik kan I række ud til Christian Lüdemann. Giv lyd hvis du har brug for andet fra mig.

**Comment by jkaltoft** [comment]
**Pre-tech refinement considerations**

Overall there are two general approaches to a solution:

1. Patch only where it is needed
    - Add print styles to override screen styles in specific places
2. Separate screen and print styles completely and build print styles from scratch
    - E.g. put all existing screen styles in a media query targeting media type `screen`
    - Create generic base print styles and specific print styles for *all* Kirby components

**Approach 1)**

*Benefits*

- We only add print styles where we need them so we may have to do less work for now

*Drawbacks*

- Screen styles will "bleed" into print as well (this may also be exactly what we want)
- We may need to add or change print styles whenever we add or modify components
    - Changes in screen styles may cause print styles to break
    - Print styles are typically not the primary focus of development and is at risk of being forgotten


**Approach 2)**

*Benefits*

- Print styles can be handled independently of screen styles
    - Better control
- Generic base print styles and specific print styles for every Kirby component could prove to be more future friendly for developers
    - The risk that future developers (including ourselves) will mess up print styles when editing screen styles is completely reduced

*Drawbacks*

- We may have to do more work up-front (but less in the long run)
- We may have to (re-)create styles that could be useful for both screen and print
    - The amount of styles that are useful for both media types may be limited
- It may be a somewhat difficult task to disable screen styles since they're applied through Javascript
    - Especially styles originating from Ionic

**Comment by RasmusKjeldgaard** [comment]
Basic print styles were added in https://github.com/kirbydesign/designsystem/pull/1667, please open a new issue and refer this one if additional print styles are needed.
